src/snovault/schema_views.py

```diff
@@ -81,3 +82,16 @@ def schemas_map(context, request):
             profiles_map[type_info.name] = type_info.schema['id']
     profiles_map['@type'] = ['JSONSchemas']
     return profiles_map
+
+
+@view_config(route_name='schemas_titles', request_method='GET',
+             decorator=etag_app_version_effective_principals)
+def schemas_titles(context, request): # pylint: disable=unused-argument
+    '''Return mapping of all schema @types and their corresponding titles'''
+    types = request.registry[TYPES]
+    profiles_titles = {}
```

Comment:
This could be dict comprehension
```
profiles_titles = {
    type_info.name, type_info.schema['title']
    for type_info in types.by_item_type.values()
    if 'title' in type_info.schema
}
```